src/consensus/aft/raft.h

```diff
@@ -311,14 +326,16 @@ namespace aft
           CCF_ASSERT(
             state->retirement_phase == kv::RetirementPhase::Completed,
             "Node is not retired, cannot become retired committed");
-          CCF_ASSERT_FMT(
-            state->retired_committed_idx == state->commit_idx,
-            "Retired "
-            "committed index {} does not match current commit index {}",
-            state->retired_committed_idx.value_or(0),
-            state->commit_idx);
-          state->retirement_phase = kv::RetirementPhase::RetiredCommitted;
           state->retired_committed_idx = seqno;
+          become_retired(seqno, kv::RetirementPhase::RetiredCommitted);
+        }
+        else
+        {
+          // Once a node's retired_committed status is itself committed, all
+          // future primaries in the network must be aware its retirement is
+          // committed, and so no longer need any communication with it to
+          // advance commit. No further communication with this node is needed.
+          all_other_nodes.erase(node_id);
```

Comment:
We should log when we pop nodes from here.

Reply:
Agreed, done. Deliberately keeping the log fairly terse until subsequent rename.